### PR TITLE
Adds scope and scope verification to SDK.

### PR DESCRIPTION
**Pull request by obber**
As title.

* Exports the `FractalSdkScope` enum for consumption
* Adds a `scope` prop to `<SignInWithFractal>`
* Adds unit tests for scope value verification (in case caller is using JS)

### CODE REVIEW

File: src/components/sign-in.tsx [hunk 1]
```diff
@@ -46,19 +57,31 @@ export function UserContextProvider({ children }: UserContextProviderProps) {
...
+  scope = DEFAULT_SCOPE,
+}: SignInProps) {
   const { setUser, user } = useContext(UserContext);
   const [url, setUrl] = useState<string | undefined>();
   const [code, setCode] = useState<string | undefined>();
 
+  if (!verifyScopes(scope)) {
+    console.error(
+      'Invalid scopes. Be sure to pass in at least one of the values from ' +
+        'types/Scope.ts. Defaulting to `[Scope.IDENTIFY]`.',
```
Comment (dborstelmann) [inline src/components/sign-in.tsx:73]: lowercase s?

File: src/components/sign-in.tsx [hunk 2]
```diff
@@ -46,19 +57,31 @@ export function UserContextProvider({ children }: UserContextProviderProps) {
...
+
   useEffect(() => {
     const getUrl = async () => {
       try {
         const urlInfo = (
           await authApiClient.v2.getUrl({
             clientId,
-            // TODO(Kan): Allow configuring these permissions.
-            scope: ['items:read', 'coins:read', 'identify'],
+            scope: scope as string[],
```
Comment (dborstelmann) [inline src/components/sign-in.tsx:84]: why do you need an as?

**Review by dborstelmann — APPROVED** [review]

File: src/components/sign-in.tsx (hunk 2 above)
Comment (obber) [reply src/components/sign-in.tsx:84]: Done.

File: src/components/sign-in.tsx (hunk 1 above)
Comment (obber) [reply src/components/sign-in.tsx:73]: Done.

**Pull request merged**